### PR TITLE
Simplify F.softmax test

### PR DESCRIPTION
**Pull request by aksub99**
Simplified F.softmax test rewritten using testing.FunctionTestCase.
Ref #6071

### CODE REVIEW

**Comment by aksub99** [comment]
@Crissman Could you please review the code and suggest relevant changes? I'm not able to pinpoint the root cause of the shape mismatch error. Thanks in advance!

File: tests/chainer_tests/functions_tests/activation_tests/test_softmax.py [hunk 1]
```diff
@@ -26,105 +25,55 @@
...
+        x, = inputs
+        return functions.softmax(x, axis=self.axis),
+
+    def forward_expected(self, inputs):
+        x, = inputs
+        expected = numpy.exp(x)
+        expected = numpy.rollaxis(expected, self.axis, expected.ndim)
+        for i in numpy.ndindex(expected.shape[:-1]):
+            expected[i] /= expected[i].sum()
+        return expected.astype(x.dtype),
```
Comment (Crissman) [inline tests/chainer_tests/functions_tests/activation_tests/test_softmax.py]: After changing the axis with `rollaxis`, it needs to be returned to the original shape after the `ndindex` is iterated.

```expected = numpy.rollaxis(expected, expected.ndim - 1, self.axis)                                                                                                                                                                                             ```

Comment (aksub99) [reply tests/chainer_tests/functions_tests/activation_tests/test_softmax.py]: Thanks for the review @Crissman . Made the fixes! Shapes are now matching.

Comment (aksub99) [reply tests/chainer_tests/functions_tests/activation_tests/test_softmax.py]: Thanks @Crissman . Made the required changes! The shapes are compatible now.

**Comment by aksub99** [comment]
I am facing issues while committing the changes to this PR because of some problems in my fork. So, I've opened a new PR #6371 with the required fixes. Sorry for the inconvenience caused.

**Comment by Crissman** [comment]
Closing since re-issued as PR #6371 .
